cmd/manager/v1alpha2/main.go

```diff
@@ -24,93 +25,142 @@ type server struct {
-	return nil, nil
+func (s *server) UpdateTrialStatus(ctx context.Context, in *api_pb.UpdateTrialStatusRequest) (*api_pb.UpdateTrialStatusReply, error) {
+	err := dbIf.UpdateTrialStatus(in.TrialName, in.NewStatus)
+	return &api_pb.UpdateTrialStatusReply{}, err
 }
 
 // Report a log of Observations for a Trial.
 // The log consists of timestamp and value of metric.
 // Katib store every log of metrics.
 // You can see accuracy curve or other metric logs on UI.
-func (s *server) ReportObservationLog(context.Context, *api_pb.ReportObservationLogRequest) (*api_pb.ReportObservationLogReply, error) {
-	return nil, nil
+func (s *server) ReportObservationLog(ctx context.Context, in *api_pb.ReportObservationLogRequest) (*api_pb.ReportObservationLogReply, error) {
+	err := dbIf.RegisterObservationLog(in.TrialName, in.ObservationLog)
+	return &api_pb.ReportObservationLogReply{}, err
 }
 
 // Get all log of Observations for a Trial.
-func (s *server) GetObservationLog(context.Context, *api_pb.GetObservationLogRequest) (*api_pb.GetObservationLogReply, error) {
-	return nil, nil
+func (s *server) GetObservationLog(ctx context.Context, in *api_pb.GetObservationLogRequest) (*api_pb.GetObservationLogReply, error) {
+	ol, err := dbIf.GetObservationLog(in.TrialName, in.StartTime, in.EndTime)
+	return &api_pb.GetObservationLogReply{
+		ObservationLog: ol,
+	}, err
 }
 
 // Get Suggestions from a Suggestion service.
-func (s *server) GetSuggestions(context.Context, *api_pb.GetSuggestionsRequest) (*api_pb.GetSuggestionsReply, error) {
-	return nil, nil
+func (s *server) GetSuggestions(ctx context.Context, in *api_pb.GetSuggestionsRequest) (*api_pb.GetSuggestionsReply, error) {
+	if in.AlgorithmName == "" {
+		return &api_pb.GetSuggestionsReply{Trials: []*api_pb.Trial{}}, errors.New("No algorithm name is specified")
+	}
+	conn, err := grpc.Dial("vizier-suggestion-"+in.AlgorithmName+":6789", grpc.WithInsecure())
+	if err != nil {
+		return &api_pb.GetSuggestionsReply{Trials: []*api_pb.Trial{}}, err
+	}
+
+	defer conn.Close()
+	c := api_pb.NewSuggestionClient(conn)
+	r, err := c.GetSuggestions(ctx, in)
+	if err != nil {
+		return &api_pb.GetSuggestionsReply{Trials: []*api_pb.Trial{}}, err
+	}
+	return r, nil
 }
 
 // Validate AlgorithmSettings in an Experiment.
 // Suggestion service should return INVALID_ARGUMENT Error when the parameter is invalid
-func (s *server) ValidateAlgorithmSettings(context.Context, *api_pb.ValidateAlgorithmSettingsRequest) (*api_pb.ValidateAlgorithmSettingsReply, error) {
-	return nil, nil
+func (s *server) ValidateAlgorithmSettings(ctx context.Context, in *api_pb.ValidateAlgorithmSettingsRequest) (*api_pb.ValidateAlgorithmSettingsReply, error) {
+	if in.AlgorithmName == "" {
+		return &api_pb.ValidateAlgorithmSettingsReply{}, errors.New("No algorithm name is specified")
+	}
+	conn, err := grpc.Dial("vizier-suggestion-"+in.AlgorithmName+":6789", grpc.WithInsecure())
```

Comment:
we can add a function like `getSuggestionServiceUrl(name String) string` or even a function like `getSuggestionServiceConnection(name String)`

Reply:
done